src/ai/backend/common/types.py

```diff
@@ -411,21 +411,55 @@ class SessionTypes(enum.StrEnum):
     INFERENCE = "inference"
     SYSTEM = "system"
 
+    @classmethod
+    def from_str(cls, s: str) -> SessionTypes:
+        match s.upper():
+            case "INTERACTIVE":
+                return cls.INTERACTIVE
+            case "BATCH":
+                return cls.BATCH
+            case "INFERENCE":
+                return cls.INFERENCE
+            case "SYSTEM":
+                return cls.SYSTEM
+            case _:
+                raise ValueError(f"Invalid session type: {s}")
+
 
 class SessionResult(enum.StrEnum):
     UNDEFINED = "undefined"
     SUCCESS = "success"
     FAILURE = "failure"
 
+    @classmethod
+    def from_str(cls, s: str) -> SessionResult:
+        match s.upper():
+            case "UNDEFINED":
+                return cls.UNDEFINED
+            case "SUCCESS":
+                return cls.SUCCESS
+            case "FAILURE":
+                return cls.FAILURE
+        raise ValueError(f"Invalid session result: {s}")
+
 
 class ResourceGroupType(enum.StrEnum):
     COMPUTE = enum.auto()
     STORAGE = enum.auto()
 
 
 class ClusterMode(enum.StrEnum):
-    SINGLE_NODE = "single-node"
-    MULTI_NODE = "multi-node"
+    SINGLE_NODE = "SINGLE_NODE"
+    MULTI_NODE = "MULTI_NODE"
+
+    @classmethod
+    def from_str(cls, s: str) -> ClusterMode:
+        match s.upper():
+            case "SINGLE_NODE":
+                return cls.SINGLE_NODE
+            case "MULTI_NODE":
+                return cls.MULTI_NODE
```

Comment:
This part would be incompatible with the old and upper alone, so why the change in behavior?